pkg/security/module/cws.go

```diff
@@ -323,47 +332,69 @@ func (c *CWSConsumer) LoadPolicies(policyProviders []rules.PolicyProvider, sendL
 	// load policies
 	c.policyLoader.SetProviders(policyProviders)
 
-	// standard ruleset
-	ruleSet := c.probe.NewRuleSet(c.getEventTypeEnabled())
+	evaluationSet, err := c.probe.NewEvaluationSet(c.getEventTypeEnabled(), []string{ProbeEvaluationRuleSetTagValue, ThreatScoreRuleSetTagValue})
+	if err != nil {
+		return err
+	}
 
-	loadErrs := ruleSet.LoadPolicies(c.policyLoader, c.policyOpts)
+	loadErrs := evaluationSet.LoadPolicies(c.policyLoader, c.policyOpts)
 	if loadErrs.ErrorOrNil() != nil {
 		logLoadingErrors("error while loading policies: %+v", loadErrs)
 	}
 
 	// update current policies related module attributes
-	c.policiesVersions = getPoliciesVersions(ruleSet)
+	c.policiesVersions = getPoliciesVersions(evaluationSet)
 	c.policyProviders = policyProviders
-	c.currentRuleSet.Store(ruleSet)
 
 	// notify listeners
 	if c.rulesLoaded != nil {
-		c.rulesLoaded(ruleSet, loadErrs)
+		c.rulesLoaded(evaluationSet, loadErrs)
 	}
 
-	// add module as listener for ruleset events
-	ruleSet.AddListener(c)
-
-	// analyze the ruleset, push default policies in the kernel and generate the policy report
-	report, err := c.probe.ApplyRuleSet(ruleSet)
-	if err != nil {
-		return err
+	// add module as listener for rule match callback
+	for _, rs := range evaluationSet.RuleSets {
+		rs.AddListener(c)
 	}
-	c.displayApplyRuleSetReport(report)
-
-	// set the rate limiters
-	c.rateLimiter.Apply(ruleSet, events.AllCustomRuleIDs())
 
 	// full list of IDs, user rules + custom
 	var ruleIDs []rules.RuleID
-	ruleIDs = append(ruleIDs, ruleSet.ListRuleIDs()...)
 	ruleIDs = append(ruleIDs, events.AllCustomRuleIDs()...)
 
+	var probeEvaluationRuleSet, threatScoreRuleSet *rules.RuleSet
+	if _, ok := evaluationSet.RuleSets[ProbeEvaluationRuleSetTagValue]; ok {
+		probeEvaluationRuleSet = evaluationSet.RuleSets[ProbeEvaluationRuleSetTagValue]
+	}
+	if _, ok := evaluationSet.RuleSets[ThreatScoreRuleSetTagValue]; ok {
+		threatScoreRuleSet = evaluationSet.RuleSets[ThreatScoreRuleSetTagValue]
+	}
+
+	if threatScoreRuleSet != nil {
+		c.currentThreatScoreRuleSet.Store(threatScoreRuleSet)
+		ruleIDs = append(ruleIDs, threatScoreRuleSet.ListRuleIDs()...)
+	}
+
+	if probeEvaluationRuleSet != nil {
+		ruleIDs = append(ruleIDs, probeEvaluationRuleSet.ListRuleIDs()...)
```

Comment:
nit:
```
c.currentRuleSet.Store(probeEvaluationRuleSet)
ruleIDs = append(ruleIDs, probeEvaluationRuleSet.ListRuleIDs()...)
```
so that it looks the same than above